### PR TITLE
ndn-cxx: 0.6.3 -> 0.7.1

### PR DESCRIPTION
**Pull request by prusnak**
###### Motivation for this change

Version update, package cleanup (part of https://github.com/NixOS/nixpkgs/issues/101964)

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/ndn-cxx/default.nix [hunk 1]
```diff
@@ -1,24 +1,36 @@
...
   pname = "ndn-cxx";
-  inherit version;
+  version = "0.7.1";
+
   src = fetchFromGitHub {
     owner = "named-data";
     repo = "ndn-cxx";
-    rev = "a3bf4319ed483a4a6fe2c96b79ec4491d7217f00";
-    sha256 = "076jhrjigisqz5n8dgxwd5fhimg69zhm834m7w9yvf9afgzrr50h";
+    rev = "${pname}-${version}";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/ndn-cxx/default.nix:20]: Isn't that $name or was that combined with ``_``?

File: pkgs/development/libraries/ndn-cxx/default.nix [hunk 2]
```diff
@@ -1,24 +1,36 @@
...
+    sha256 = "1lcaqc79n3d9sip7knddblba17sz18b0w7nlxmj3fz3lb3z9qd51";
   };
+
   nativeBuildInputs = [ pkgconfig wafHook ];
-  buildInputs = [ openssl doxygen boost sqlite python pythonPackages.sphinx];
+
+  buildInputs = [ openssl doxygen boost sqlite python3 python3Packages.sphinx ];
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/ndn-cxx/default.nix:26]: Is sphinx a runtime dependency? I thought it only generates docs.

File: pkgs/development/libraries/ndn-cxx/default.nix (hunk 1 above)
Comment (prusnak) [reply pkgs/development/libraries/ndn-cxx/default.nix:20]: I want to keep this explicit for GitHub revision

File: pkgs/development/libraries/ndn-cxx/default.nix (hunk 2 above)
Comment (prusnak) [reply pkgs/development/libraries/ndn-cxx/default.nix]: Moved to nativeBuildInputs (together with python3 and doxygen) in 50bf5d66741919888d9fb72a0c7b6633a9969629

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 102248` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ndn-cxx</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 102248` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>ndn-cxx</li>
  </ul>
</details>

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 102248 [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>1 package built:</summary>  <ul>    <li>ndn-cxx</li>  </ul></details>

**Pull request merged**